src/v/cluster/node_status_backend.h

```diff
@@ -47,6 +47,10 @@ namespace cluster {
 class node_status_backend {
 public:
     static constexpr ss::shard_id shard = 0;
+    // after this many consecutive timeouts, reset the connection
+    static constexpr uint8_t consecutive_request_timeouts = 20u;
+    // logging rate limit for potentially noisy logs
+    static constexpr auto rate_limit = std::chrono::seconds(1);
```

Comment:
nit: make constants private (preferably move to cc file, don't think these are of any use in the header)

isn't this rate limit technically worse than no ratelimit? 

originally it was 20 times 100ms apart, once every 2s.. make it 10s or something?